src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -238,6 +263,127 @@ CHIP_ERROR GeneralDiagosticsAttrAccess::Read(const ConcreteReadAttributePath & a
     return CHIP_NO_ERROR;
 }
 
+class GeneralDiagnosticsCommandHandler : public CommandHandlerInterface
+{
+public:
+    // Register for the GeneralDiagnostics cluster on all endpoints.
+    GeneralDiagnosticsCommandHandler() : CommandHandlerInterface(Optional<EndpointId>::Missing(), GeneralDiagnostics::Id) {}
+
+    void InvokeCommand(HandlerContext & handlerContext) override;
+
+private:
+    void HandleTestEventTrigger(HandlerContext & ctx, const Commands::TestEventTrigger::DecodableType & commandData);
+    void HandleTimeSnapshot(HandlerContext & ctx, const Commands::TimeSnapshot::DecodableType & commandData);
+    void HandlePayloadTestRequest(HandlerContext & ctx, const Commands::PayloadTestRequest::DecodableType & commandData);
+};
+
+GeneralDiagnosticsCommandHandler gCommandHandler;
+
+void GeneralDiagnosticsCommandHandler::InvokeCommand(HandlerContext & handlerContext)
```

Comment:
There seems to be a pretty nonzero codesize cost here.  What can we do to improve that?

Reply:
Now, flash increase is negative